### PR TITLE
Fix TP

### LINKED ISSUE #16: CUDA initialization / operation error when using distributed GPU
* When I run `test_lmcache_local_distributed()`, `test_lmcache_cachegen_distributed()`, and `test_lmcache_safetensor_distributed()` in `tests/tests.py`, there is the following error message in `stdout.log`: `RuntimeError: CUDA error: initialization error`.
* According to an issue on initializing CUDA with `tensor_parallel_size>1` with vllm (https://github.com/vllm-project/vllm/issues/6152), I set `VLLM_WORKER_MULTIPROC_METHOD=spawn`. Then I get the following error message in `stderr.log`:

  ```shell
  [33mERROR LMCache: [0mEngine background task failed [2024-10-15 00:59:31,317.317] [line: 188] [file: vllm_injection.py]
  Traceback (most recent call last):
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/vllm/engine/async_llm_engine.py", line 727, in run_engine_loop
      done, _ = await asyncio.wait(
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/asyncio/tasks.py", line 384, in wait
      return await _wait(fs, timeout, return_when, loop)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/asyncio/tasks.py", line 491, in _wait
      await waiter
  asyncio.exceptions.CancelledError
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "/local/shaotingf/lmcache1/lmcache-vllm/lmcache_vllm/vllm_injection.py", line 177, in new_log_task_completion
      return_value = task.result()
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/vllm/engine/async_llm_engine.py", line 726, in run_engine_loop
      async with asyncio_timeout(ENGINE_ITERATION_TIMEOUT_S):
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/vllm/engine/async_timeout.py", line 95, in __aexit__
      self._do_exit(exc_type)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/vllm/engine/async_timeout.py", line 178, in _do_exit
      raise asyncio.TimeoutError
  asyncio.exceptions.TimeoutError
  [33mERROR LMCache: [0mException in callback functools.partial(<function new_log_task_completion at 0x7ff74922f250>, error_callback=<bound method AsyncLLMEngine._error_callback of <vllm.engine.async_llm_engine.AsyncLLMEngine object at 0x7ff7442250f0>>)
  handle: <Handle functools.partial(<function new_log_task_completion at 0x7ff74922f250>, error_callback=<bound method AsyncLLMEngine._error_callback of <vllm.engine.async_llm_engine.AsyncLLMEngine object at 0x7ff7442250f0>>)> [2024-10-15 00:59:31,320.320] [line: 82] [file: __init__.py]
  Traceback (most recent call last):
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/vllm/engine/async_llm_engine.py", line 727, in run_engine_loop
      done, _ = await asyncio.wait(
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/asyncio/tasks.py", line 384, in wait
      return await _wait(fs, timeout, return_when, loop)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/asyncio/tasks.py", line 491, in _wait
      await waiter
  asyncio.exceptions.CancelledError
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "/local/shaotingf/lmcache1/lmcache-vllm/lmcache_vllm/vllm_injection.py", line 177, in new_log_task_completion
      return_value = task.result()
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/vllm/engine/async_llm_engine.py", line 726, in run_engine_loop
      async with asyncio_timeout(ENGINE_ITERATION_TIMEOUT_S):
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/vllm/engine/async_timeout.py", line 95, in __aexit__
      self._do_exit(exc_type)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/vllm/engine/async_timeout.py", line 178, in _do_exit
      raise asyncio.TimeoutError
  asyncio.exceptions.TimeoutError
  
  The above exception was the direct cause of the following exception:
  
  Traceback (most recent call last):
    File "uvloop/cbhandles.pyx", line 63, in uvloop.loop.Handle._run
    File "/local/shaotingf/lmcache1/lmcache-vllm/lmcache_vllm/vllm_injection.py", line 190, in new_log_task_completion
      raise RuntimeError(
  RuntimeError: Task finished unexpectedly. This should never happen! Please open an issue on Github. See stack trace above for the actual cause.
  ERROR:    Exception in ASGI application
  Traceback (most recent call last):
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/responses.py", line 257, in __call__
      await wrap(partial(self.listen_for_disconnect, receive))
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/responses.py", line 253, in wrap
      await func()
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/responses.py", line 230, in listen_for_disconnect
      message = await receive()
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 555, in receive
      await self.message_event.wait()
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/asyncio/locks.py", line 214, in wait
      await fut
  asyncio.exceptions.CancelledError: Cancelled by cancel scope 7fdc925c38b0
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 401, in run_asgi
      result = await app(  # type: ignore[func-returns-value]
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 60, in __call__
      return await self.app(scope, receive, send)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/fastapi/applications.py", line 1054, in __call__
      await super().__call__(scope, receive, send)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/applications.py", line 113, in __call__
      await self.middleware_stack(scope, receive, send)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/middleware/errors.py", line 187, in __call__
      raise exc
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/middleware/errors.py", line 165, in __call__
      await self.app(scope, receive, _send)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/middleware/cors.py", line 85, in __call__
      await self.app(scope, receive, send)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
      await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/_exception_handler.py", line 62, in wrapped_app
      raise exc
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/_exception_handler.py", line 51, in wrapped_app
      await app(scope, receive, sender)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/routing.py", line 715, in __call__
      await self.middleware_stack(scope, receive, send)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/routing.py", line 735, in app
      await route.handle(scope, receive, send)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/routing.py", line 288, in handle
      await self.app(scope, receive, send)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/routing.py", line 76, in app
      await wrap_app_handling_exceptions(app, request)(scope, receive, send)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/_exception_handler.py", line 62, in wrapped_app
      raise exc
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/_exception_handler.py", line 51, in wrapped_app
      await app(scope, receive, sender)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/routing.py", line 74, in app
      await response(scope, receive, send)
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/starlette/responses.py", line 250, in __call__
      async with anyio.create_task_group() as task_group:
    File "/local/shaotingf/anaconda3/envs/lmcache1/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 736, in __aexit__
      raise BaseExceptionGroup(
  exceptiongroup.ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
  ```

### PR DESCRIPTION
**Pull request by YaoJiayi**
This PR resolves https://github.com/LMCache/lmcache-tests/issues/16

### CODE REVIEW

File: lmcache_vllm/vllm_adapter.py [hunk 1]
```diff
@@ -85,6 +87,27 @@ def init_lmcache_engine(
 
     return engine
 
+def broadcast_seq_group_metadata(
+    model_input,
+    is_driver_worker,
+):
```
Comment (ApostaC) [inline lmcache_vllm/vllm_adapter.py]: Can you add a very brief docstring for this function?

**Review by ApostaC — DISMISSED** [review]
LGTM! Just one minor comment. Give an approve so that you can merge after fixing the comment.

File: lmcache_vllm/vllm_adapter.py (hunk 1 above)
Comment (YaoJiayi) [reply lmcache_vllm/vllm_adapter.py]: Sure.

**Review by ApostaC — APPROVED** [review]
Looks good to me, feel free to merge!

**Pull request merged**